src/Core/System.php

```diff
@@ -222,28 +230,25 @@ public static function currentLoad()
 	 * @param string $url  The new Location to redirect
 	 * @param int    $code The redirection code, which is used (Default is 302)
 	 *
-	 * @throws InternalServerErrorException If the URL is not fully qualified
+	 * @throws BadRequestException If the URL is not fully qualified
 	 */
 	public static function externalRedirect($url, $code = 302)
 	{
 		if (empty(parse_url($url, PHP_URL_SCHEME))) {
-			throw new InternalServerErrorException("'$url' is not a fully qualified URL, please use App->internalRedirect() instead");
+			throw new BadRequestException("'$url' is not a fully qualified URL, please use App->internalRedirect() instead");
```

Comment:
Technically it should be a native `\InvalidArgumentException` which falls within the purview of Friendica's `InternalServerErrorException` because it is code-related, not related to the HTTP request.

Reply:
Well, I haven't understood why there is this exception in this place. (and why we have to handle this situation anyway).

Reply:
It's a debugging exception to figure out where we wrongly generate a URL for redirection.

Reply:
Are you planning on reverting this change?

Reply:
I would like to avoid `InternalServerErrorException` and to be more precise with the messages to easier separate these errors between core and web server.

Reply:
This is an `InternalServerErrorException`, it's unlikely any change in the request would lead to a different outcome.

Reply:
I have got several issues with this exception. At first: this situation is fixable. When there is no host name then it is an internal redirect. Also it targets the wrong people. It is meant for the programmers but is directed to the users. And it isn't logged at all. A better way would be to log this situation and then simply perform an internal redirect. From my point of view we don't need to halt operation here at all.

Reply:
The problem is that the `internalRedirect` requires the `App` context which we don't have in this low-level static method, and I don't think we should throw it in. I'm fine with logging the technical error message and showing a generic error message to the user, but there isn't much else to do in this context without invoking way more scope than intended.

Reply:
It's just `baseUrl` that we are using in that class anyway at some other place.

Reply:
I'm glad you see it that way, it was the reason @nupplaphil created the `DI` shorthands, under the hood it's a little more complex but it doesn't need to be revisited at this time. 🙂